src/winforms/toga_winforms/widgets/canvas.py

```diff
@@ -51,6 +51,17 @@ def matrix(self):
     def matrix(self, matrix):
         self._matrix = matrix
 
+    @property
+    def last_point(self):
+        return self._last_point
+
+    @last_point.setter
+    def last_point(self, last_point):
+        if last_point is None:
+            self._last_point = None
+            return
+        self._last_point = tuple(map(float, last_point))
```

Comment:
Given that the issue is with the usage by the Pythonnet API, why not do this last this conversion at the point where it's being consumed by the Pythonnet API, rather than setting up this mildly complex generic accessor? i.e., why not make this a one line `PointF(float(last_point[0]), float(last_point[1]))` on L184 and L195 where it is being used?

Reply:
I made the change here so I `last_point` will not contain `np.float64` values. Otherwise, I will need to convert it everywhere to float, and it's much more lines of code.

The reason I used `tuple(map(float, last_point))` is that it does not assume the length of `last_point`. In that way if one day `last_point` will be 3D for whatever reason, this line of code won't change. It is also shorter, so that's a bonus for me.

If you which me to revert `last_point` to how it was before and convert it to `float` where it is needed, it's a no brainer. Just let me know which fix would you want me to do.

Reply:
My inclination is that `np.float64` is a valid value - *except* for passing to Python.net APIs - and even then, it's a bug that is on track to be fixed. 

Doing a more thorough inspection of the code, there's also a number of other pre-existing casts of `last_point`; interestingly, a lot of those are casting to `int`, not `float`. 

Since we're likely going to be mixing uses of `last_point` and directly user-provided arguments, my inclination is to audit the code so that every call to a Python.net API does a cast to an appropriate type (including auditing the existing casts to `int` to see if they should be `float` casts), and do that cast as close to the Python.net API as possible. If we made `last_point` "smart", we'd have a mix of explicit and implicit casts in the code (as observed at the point of invoking a Python.net API), which will ultimately be confusing.

Reply:
Done.